File: app/src/main/java/com/xuie/android/provider/ColorContract.java
package com.xuie.android.provider;

import android.content.ContentUris;
import android.net.Uri;
import android.provider.BaseColumns;

/**
 *
 * @author xuie
 * @date 2017/4/12 0012
 */

public class ColorContract {
    /**
     * Content authority is a name for the entire content provider
     * similar to a domain and its website. This string is guaranteed to be unique.
     */
    public static final String CONTENT_AUTHORITY = "com.xuie.android.data.content.provider";

    /**
     * Use the content authority to provide the base
     * of all URIs
     */
    private static final Uri BASE_CONTENT_URI = Uri.parse("content://" + CONTENT_AUTHORITY);

    /**
     * Path for URI to a color
     */
    public static final String PATH_COLOR = "color";

    /**
     * Class that defines the schema of the Color table.
     */
    public static final class ColorEntry implements BaseColumns {
        /**
         * Uri to access all movies
         */
        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_COLOR).build();

        /**
         * MIME types of Color queries
         */
        public static final String CONTENT_TYPE =
                "vnd.android.cursor.dir/" + CONTENT_URI + "/" + PATH_COLOR;
        public static final String CONTENT_ITEM_TYPE =
                "vnd.android.cursor.item/" + CONTENT_URI + "/" + PATH_COLOR;

        /**
         * Schema
         */
        public static final String TABLE_NAME = "color_database";
        public static final String COLUMN_NAME = "name";
        public static final String COLUMN_COLOR = "color";

        /**
         * Builds a URI for an individual movie.
         */
        public static Uri buildUri(long id) {
            return ContentUris.withAppendedId(CONTENT_URI, id);
        }
    }

}

File: app/src/main/java/com/xuie/android/provider/ColorContract.kt
package com.xuie.android.provider

import android.content.ContentUris
import android.net.Uri
import android.provider.BaseColumns

/**
 * @author xuie
 * @date 2017/4/12 0012
 */

object ColorContract {
  /**
   * Content authority is a name for the entire content provider
   * similar to a domain and its website. This string is guaranteed to be unique.
   */
  val CONTENT_AUTHORITY = "com.xuie.android.data.content.provider"

  /**
   * Use the content authority to provide the base
   * of all URIs
   */
  private val BASE_CONTENT_URI = Uri.parse("content://$CONTENT_AUTHORITY")

  /**
   * Path for URI to a color
   */
  val PATH_COLOR = "color"

  /**
   * Class that defines the schema of the Color table.
   */
  class ColorEntry : BaseColumns {
    companion object {
      /**
       * Uri to access all movies
       */
      val CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_COLOR).build()

      /**
       * MIME types of Color queries
       */
      val CONTENT_TYPE =
          "vnd.android.cursor.dir/$CONTENT_URI/$PATH_COLOR"
      val CONTENT_ITEM_TYPE =
          "vnd.android.cursor.item/$CONTENT_URI/$PATH_COLOR"

      /**
       * Schema
       */
      val TABLE_NAME = "color_database"
      val COLUMN_NAME = "name"
      val COLUMN_COLOR = "color"

      /**
       * Builds a URI for an individual movie.
       */
      fun buildUri(id: Long): Uri {
        return ContentUris.withAppendedId(CONTENT_URI, id)
      }
    }
  }

}
